pkg/auth/cluster/context.go

```diff
@@ -30,18 +30,26 @@ func StreamAuthorizedID(ctx context.Context) string {
 	return ctx.Value(ClusterIDKey).(string)
 }
 
-func AuthorizedOutgoingContext(ctx context.Context) context.Context {
-	return metadata.AppendToOutgoingContext(ctx, string(ClusterIDKey), StreamAuthorizedID(ctx))
+func (k clusterIDKeyType) GetRequestMetadata(ctx context.Context, _ ...string) (map[string]string, error) {
+	if v := ctx.Value(k); v != nil {
+		return map[string]string{
+			string(k): v.(string),
+		}, nil
+	}
+	return nil, nil
+}
+
+func (k clusterIDKeyType) RequireTransportSecurity() bool {
+	return true
 }
 
-func AuthorizedIDFromIncomingContext(ctx context.Context) (string, bool) {
+func (k clusterIDKeyType) FromIncomingCredentials(ctx context.Context) context.Context {
```

Comment:
This will need to be updated in https://github.com/rancher/opni/blob/main/plugins/logging/pkg/otel/otel.go#L139 as well as the others that you have updated